apps/antalmanac/src/components/RightPane/SectionTable/EnrollmentHistoryPopup.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect, useMemo } from 'react';
+import { LineChart, Line, CartesianGrid, ResponsiveContainer, XAxis, YAxis, Tooltip, Legend } from 'recharts';
+import { Box, Link, Typography, Skeleton } from '@mui/material';
+import EnrollmentHistoryHelper, { EnrollmentHistory } from '$lib/enrollmentHistory';
+import { isDarkMode } from '$lib/helpers';
+
+export interface EnrollmentHistoryPopupProps {
+    department: string;
+    courseNumber: string;
+    isMobileScreen: boolean;
+}
+
+const EnrollmentHistoryPopup = (props: EnrollmentHistoryPopupProps) => {
+    const { department, courseNumber, isMobileScreen } = props;
+    const [loading, setLoading] = useState(true);
+    const [enrollmentHistory, setEnrollmentHistory] = useState<EnrollmentHistory>();
+
+    const graphWidth = useMemo(() => (isMobileScreen ? 250 : 450), [isMobileScreen]);
+    const graphHeight = useMemo(() => (isMobileScreen ? 175 : 250), [isMobileScreen]);
+    const popupTitle = useMemo(() => {
+        return enrollmentHistory
+            ? `${department} ${courseNumber} | ${enrollmentHistory.year} ${
+                  enrollmentHistory.quarter
+              } | ${enrollmentHistory.instructors.join(', ')}`
+            : 'No past enrollment data found for this course';
+    }, [courseNumber, department, enrollmentHistory]);
+
+    const encodedDept = useMemo(() => encodeURIComponent(department), [department]);
+    const axisColor = isDarkMode() ? '#fff' : '#111';
+    const tooltipDateColor = '#111';
+
+    useEffect(() => {
+        if (!loading) {
+            return;
+        }
+
+        EnrollmentHistoryHelper.queryEnrollmentHistory(department, courseNumber).then((enrollmentHistory) => {
+            if (enrollmentHistory) {
+                setEnrollmentHistory(enrollmentHistory);
```

Comment:
This just looks so confusing when browsing through. I think you should rename `enrollmentHistory` here to `data` or `enrollmentHistory_`, which is not elegant but at least doesn't mirror an outside declaration

Reply:
Thanks for catching that, I renamed it to `data` just to keep it short